web/client/utils/LayersUtils.js

```diff
@@ -134,6 +134,26 @@ const deepChange = (nodes, findValue, propName, propValue) => {
     return [];
 };
 
+/**
+ * adds or update node property in a nested node
+ * similar to deepchange but it overrides a whole group of options instead of one
+*/
+const deepOptionsChange = (nodes, findId, propValues = {}) => {
```

Comment:
I think this can be implemented by iterating on propValues keys for deepChange (or vice-versa deepChange can be implemented using this). There is a little of redundancy.
It's the function enough tested to replace the implementation? otherwise you can skip, maybe comment